internal/app/synchronizer/synchronizer_service_test.go

```diff
@@ -13,3 +13,221 @@
+	tm := []*pb.Match{}
+	for i := 0; i < 30; i++ {
+		tm = append(tm, &pb.Match{MatchId: xid.New().String()})
+	}
+
+	testCases := []*testData{
+		{
+			description: "basic evaluation scenario, multiple calls with valid results in a registration window",
+			testCalls: []*testCallData{
+				{
+					proposals:   tm[0:5],
+					wantResults: tm[0:5],
+				},
+				{
+					proposals:   tm[5:10],
+					wantResults: tm[5:10],
+				},
+			},
+			testEvaluator: &testEvaluatorData{
+				results: [][]*pb.Match{tm[0:10]},
+			},
+			regInterval:  "500ms",
+			propInterval: "500ms",
+		},
+		{
+			description: "basic evaluation sending subset of proposals in proposal acceptance window",
+			testCalls: []*testCallData{
+				{
+					proposals:     tm[0:5],
+					wantResults:   []*pb.Match{tm[0], tm[2]},
+					evaluateDelay: 200,
+				},
+				{
+					proposals:     tm[5:10],
+					wantResults:   []*pb.Match{tm[5], tm[8], tm[9]},
+					evaluateDelay: 1200,
+				},
+			},
+			testEvaluator: &testEvaluatorData{
+				results: [][]*pb.Match{{tm[0], tm[2], tm[5], tm[8], tm[9]}},
+			},
+			regInterval:  "1000ms",
+			propInterval: "1000ms",
+		},
+		{
+			description: "Evaluation proposals miss the evaluation window",
+			testCalls: []*testCallData{
+				{
+					proposals:   tm[0:5],
+					wantResults: tm[0:5],
+				},
+				{
+					evaluateDelay:      1000,
+					proposals:          tm[5:10],
+					evaluationrErrCode: codes.DeadlineExceeded,
+				},
+			},
+			testEvaluator: &testEvaluatorData{
+				results: [][]*pb.Match{tm[0:5]},
+			},
+			regInterval:  "200ms",
+			propInterval: "200ms",
+		},
+		{
+			description: "Blocking registration requests coming in during proposal acceptance",
+			testCalls: []*testCallData{
+				{
+					proposals:   tm[0:5],
+					wantResults: tm[0:5],
+				},
+				{
+					registerDelay: 1000,
+					proposals:     tm[5:10],
+					wantResults:   tm[5:10],
+				},
+			},
+			testEvaluator: &testEvaluatorData{
+				results: [][]*pb.Match{tm[0:5], tm[5:10]},
+			},
+			regInterval:  "200ms",
+			propInterval: "1000ms",
+		},
+		{
+			description: "Mix of successful requests and requests missing evaluation window",
+			testCalls: []*testCallData{
+				{
+					evaluateDelay:      3500,
+					proposals:          tm[0:5],
+					evaluationrErrCode: codes.DeadlineExceeded,
+				},
+				{
+					evaluateDelay:      3500,
+					proposals:          tm[5:10],
+					evaluationrErrCode: codes.DeadlineExceeded,
+				},
+				{
+					registerDelay: 3000,
+					evaluateDelay: 100,
+					proposals:     tm[10:15],
+					wantResults:   tm[10:15],
+				},
+				{
+					registerDelay: 3000,
+					evaluateDelay: 100,
+					proposals:     tm[15:20],
+					wantResults:   tm[15:20],
+				},
+			},
+			testEvaluator: &testEvaluatorData{
+				results: [][]*pb.Match{[]*pb.Match{}, tm[10:20]},
+			},
+			regInterval:  "2000ms",
+			propInterval: "100ms",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc.testEvaluator.eval = &testEvaluatorClient{
```

Comment:
I would prefer using
```
t.Run(tc.description, func(t *testing.T) {
...
})
```
and remove the description log at line 226 since this is the documented pattern for table testing.

Ref: https://github.com/golang/go/wiki/TableDrivenTests